kubespawner/spawner.py

```diff
@@ -1437,6 +1442,39 @@ def get_pod_manifest(self):
             logger=self.log,
         )
 
+
+    def get_secret_manifest(self):
```

Comment:
We should clean up the Secret and Service object when we kill the pod - this keeps the namespace clean, and helps lock down internal cert secrets. We can let kubernetes do the work for us with `ownerReferences` - https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/. We should add these to the secret & service.